projects/cdk/directives/auto-focus/autofocus.directive.ts

```diff
@@ -40,49 +43,70 @@ export class TuiAutoFocusDirective implements AfterViewInit {
         private readonly tuiFocusableComponent: TuiFocusableElementAccessor | null,
         @Inject(TUI_IS_IOS) private readonly isIos: boolean,
         @Inject(ANIMATION_FRAME) private readonly animationFrame$: Observable<number>,
+        @Inject(NgZone)
+        private readonly ngZone: NgZone,
     ) {}
 
     ngAfterViewInit() {
         if (!this.autoFocus) {
             return;
         }
 
-        const element =
-            this.tuiFocusableComponent?.nativeFocusableElement ||
-            this.elementRef.nativeElement;
-
         if (this.isIos) {
-            veryVerySadIosFix(this.injector);
+            this.ngZone.runOutsideAngular(() =>
+                veryVerySadIosFix(this.injector, this.element),
+            );
+        } else {
+            race(
+                timer(TIMEOUT),
+                this.animationFrame$.pipe(
+                    throttleTime(POLLING_TIME),
+                    map(() => this.element.closest(NG_ANIMATION_SELECTOR)),
+                    filter(v => !v),
+                    take(1),
+                ),
+            ).subscribe(() => setNativeFocused(this.element));
         }
+    }
 
-        race(
-            timer(TIMEOUT),
-            this.animationFrame$.pipe(
-                throttleTime(POLLING_TIME),
-                map(() => element.closest(NG_ANIMATION_SELECTOR)),
-                filter(v => !v),
-                take(1),
-            ),
-        ).subscribe(() => {
-            setNativeFocused(element);
-        });
+    private get element(): TuiNativeFocusableElement {
+        return (
+            this.tuiFocusableComponent?.nativeFocusableElement ||
+            this.elementRef.nativeElement
+        );
     }
 }
 
-function veryVerySadIosFix(injector: Injector) {
+function veryVerySadIosFix(injector: Injector, element: TuiNativeFocusableElement): void {
     const renderer = injector.get(Renderer2);
     const {nativeElement} = injector.get(ViewContainerRef).element;
-    const decoy: HTMLElement = renderer.createElement('input');
+    const fakeInput: HTMLElement = renderer.createElement('input');
 
-    decoy.style.position = 'absolute';
-    decoy.style.opacity = '0';
-    decoy.style.height = '0';
+    fakeInput.style.position = 'absolute';
+    fakeInput.style.opacity = '0';
+    fakeInput.style.height = '0';
 
-    renderer.setAttribute(decoy, 'readonly', 'readonly');
-    renderer.appendChild(nativeElement, decoy);
-    decoy.addEventListener('blur', () => renderer.removeChild(nativeElement, decoy), {
+    renderer.setAttribute(fakeInput, 'autofocus', 'true');
+    renderer.appendChild(nativeElement, fakeInput);
+
+    fakeInput.addEventListener('blur', () => setNativeFocused(fakeInput), {
         once: true,
+        passive: true,
```

Comment:
Passive makes no sense here, preventing default is impossible on `blur` event.

Reply:
fixed